app/src/main/res/layout/dialog_progress.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="horizontal"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent">
+        <ProgressBar
+            android:layout_width="@dimen/progress_bar_size"
+            android:layout_height="@dimen/progress_bar_size"
+            android:layout_gravity="center"
+            android:backgroundTint="@color/colorAccent"
+            android:padding="@dimen/padding_large" />
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:paddingRight="@dimen/padding_extreme_large"
+            android:text="Loading..."
+            android:textSize="@dimen/text_size_large" />
+
+</LinearLayout>
```

Comment:
Why do we even need this. Isn't progress dialog good enough?

Reply:
To setcontentview for dialog.

Reply:
Why do we need that?

Reply:
This is providing layout for the dialog loading dialog that pops up.

Reply:
Again, why do we need a custom layout?

Reply:
I don't know any other way of adding a progressBar to a dialog body other than making a custom layout.

Reply:
https://developer.android.com/reference/android/app/ProgressDialog

Reply:
Okay, will implement this.

Reply:
done